src/client/client.cpp

```diff
@@ -549,8 +549,8 @@ void Client::step(float dtime)
 		std::vector<v3s16> blocks_to_ack;
 		bool force_update_shadows = false;
 		auto mesh_queue_size = m_mesh_update_manager.m_queue_out.size();
-		m_avg_mesh_queue_size = MYMIN(mesh_queue_size, 0.95 * m_avg_mesh_queue_size + 0.05 * mesh_queue_size);
-		while (!m_mesh_update_manager.m_queue_out.empty() && num_processed_meshes < 0.33 * m_avg_mesh_queue_size)
+		m_avg_mesh_queue_size = MYMAX(20, 0.95 * m_avg_mesh_queue_size + 0.05 * mesh_queue_size);
```

Comment:
Honestly I would get rid of this part altogether. The point of this PR is speed up mesh generation. If now we have to slow down mesh replacement it seems pointless.

If you feel this is needed that maybe we could put it on a time budget (like 1/10 dtime or something).
But in reality I fear you'll find a setting that is suitable in all situations.

Reply:
now some nodes do not update at all